Classes/Domain/Search/ResultSet/SearchResultSetService.php

```diff
@@ -537,6 +562,83 @@ public function search(SearchRequest $searchRequest)
     }
 
     /**
+     * Allows to modify a query before eventually handing it over to Solr.
+     *
+     * @param Query $query The current query before it's being handed over to Solr.
+     * @param SearchRequest $searchRequest The searchRequest, relevant in the current context
+     * @return Query The modified query that is actually going to be given to Solr.
```

Comment:
Parameter `$search` is missing

Exception info `@throws \UnexpectedValueException ..` also